### PR TITLE
Submission of week4 day2 & 3 & 4 homework

### PR DESCRIPTION
**Pull request by kevin-hw-li**
Difficulty: 6.0

### CODE REVIEW

File: kevin_li/week04/day2/MTA.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+line_n = ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th']
+line_l = ['8th', '6th', 'Union Square', '3rd', '1st']
+line_6 = ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
+
+
+def single_line_trip(start_line, start_station, end_station)
+  if start_line.index(start_station) < start_line.index(end_station)
+    puts start_line[start_line.index(start_station)+1..start_line.index(end_station)]
+  else
+    puts start_line.reverse[start_line.reverse.index(start_station)+1..start_line.reverse.index(end_station)]
```
Comment (textchimp) [inline kevin_li/week04/day2/MTA.rb:39]: Lines like this are kind of hard to read... any way to simplify? Even if it means splitting the line over several lines?

File: kevin_li/week04/day2/MTA.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    puts end_line.reverse[end_line.reverse.index('Union Square')+1..end_line.reverse.index(end_station)]
+  end
+  puts "Total stops: #{(start_line.index(start_station) - start_line.index('Union Square')).abs + (end_line.index('Union Square') - end_line.index(end_station)).abs}"
+end
+
+
+def plan_trip(start_line, start_station, end_line, end_station)
+  puts "Travels through the stations:"
+  single_line_trip(start_line, start_station, end_station) if start_line == end_line
+  multi_line_trip(start_line, start_station, end_line, end_station) if start_line != end_line
```
Comment (textchimp) [inline kevin_li/week04/day2/MTA.rb:64]: An `if...else` structure might be more appropriate here.

**Comment by textchimp** [comment]
Tennis player DB looks good!

**Comment by kevin-hw-li** [comment]
Yes, I can simplify the codes.

**Comment by kevin-hw-li** [comment]
Thanks! I will submit it again soon.

**Pull request merged**